### PR TITLE
Added UI features

### PR DESCRIPTION
**Pull request by ujjaldas1997**
I have modified some codes, so that at run time, the program would not crash and written simple UI that run in browser having an additional dependency "flask". Feel free to discuss with me about these.

### CODE REVIEW

File: Imagehandler.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 from PatternFinding import PatternFinding
 from FindingOrientationOfContours import FindingOrientationOfContours
 from AffineTransformation import AffineTransformation
+import numpy as np
```
Comment (Griffintaur) [inline Imagehandler.py]: are we using this module in this file?

File: Imagehandler.py [hunk 2]
```diff
@@ -48,36 +49,38 @@ def WritingImage(self, image, path, imageName):
             cv.imwrite(path + imageName, image)
             cv.imshow(imageName, image)
             cv.waitKey(0)
+        cv.destroyAllWindows()
```
Comment (Griffintaur) [inline Imagehandler.py]: Alignment issue

File: Imagehandler.py [hunk 3]
```diff
@@ -48,36 +49,38 @@ def WritingImage(self, image, path, imageName):
...
-        #         cv.rectangle(self.imageOriginal,(x,y),(x+w,y+h),(0,0,255),2)
-        #         cv.imshow("hello",self.imageOriginal)
-        # maxAreaIndex=np.argmax(areas)
-        # x,y,w,h=cv.boundingRect(contours[maxAreaIndex])
-        # cv.rectangle(self.image2,(x,y),(x+w,y+h),(0,255,0),2)
-        # cv.imshow("hello",self.imageOriginal)
-        # cv.waitKey(0)
+        #uncomment this to see the contours on the image
+        #cv.drawContours(thresholdImage, contours, -1, (255,255,0), 3)
+        '''#patternFindingObj=PatternFinding()
```
Comment (Griffintaur) [inline Imagehandler.py]: ''' ?

File: Imagehandler.py [hunk 4]
```diff
@@ -48,36 +49,38 @@ def WritingImage(self, image, path, imageName):
...
         contour_group = (thresholdImage, contours, hierarchy)
         return contour_group
 
     def QRCodeInImage(self):
         patternFindingObj = PatternFinding(
             self.GetImageContour(), self.imageOriginal)
         patterns = patternFindingObj.FindingQRPatterns(3)
-        if len(patterns):
+        if len(patterns) == 0:
             print 'patterns unble to find'
```
Comment (Griffintaur) [inline Imagehandler.py]: typo

File: PatternFinding.py [hunk 5]
```diff
@@ -81,25 +83,29 @@ def FindingQRPatterns(self, nooflevels):
...
+            for i in xrange(3):                     #pick the best three
                 QRPatterns_new.append(QRPatterns[arg_areapatterns[
                     len(arg_areapatterns) - i - 1]])
                 x, y, w, h = cv.boundingRect(QRPatterns_new[i])
                 cv.rectangle(
                     self.image, (x, y), (x + w, y + h), (0, 0, 255), 2)
-                cv.imshow('hello', self.image)
+                #cv.imshow('best qr contour', self.image)           #uncomment to debug
+                #cv.waitKey(0)
+            #cv.destroyAllWindows()
```
Comment (Griffintaur) [inline PatternFinding.py]: indentation?

File: PatternFinding.py [hunk 6]
```diff
@@ -81,25 +83,29 @@ def FindingQRPatterns(self, nooflevels):
                 x, y, w, h = cv.boundingRect(self.Contours[mapping])
                 cv.rectangle(
                     self.image, (x, y), (x + w, y + h), (0, 255, 0), 2)
-                cv.imshow('hello', self.image)
+                #cv.imshow('contour of qr', self.image)         #uncomment to debug
+                #cv.waitKey(0)
                 QRPatterns.append(patterns[ind])
+            #cv.destroyAllWindows()
```
Comment (Griffintaur) [inline PatternFinding.py]: indentation?

File: config.yml [hunk 7]
```diff
@@ -4,8 +4,8 @@
 
 #@author: Ankit Singh
 Main:
-    Input : 'C:\Users\310247467\Desktop\learn\qr\Input\'
-    Output: 'C:\Users\310247467\Desktop\learn\qr\Results\'
+    Input : '/home/ujjaldas223/projects/QRCodeReader/Input/'
```
Comment (Griffintaur) [inline config.yml]: Any specific need to change this?

File: Imagehandler.py (hunk 1 above)
Comment (ujjaldas1997) [reply Imagehandler.py]: In case of debugging, line no - 70 uses numpy. But we are not using numpy.

File: Imagehandler.py (hunk 2 above)
Comment (ujjaldas1997) [reply Imagehandler.py]: If we are not using this(line no 52), then the program crashed after showing the output image.
screenshot from 2017-09-04 17-59-44

File: Imagehandler.py (hunk 3 above)
Comment (ujjaldas1997) [reply Imagehandler.py]: I am sorry, I was not aware of that change.

File: Imagehandler.py (hunk 4 above)
Comment (ujjaldas1997) [reply Imagehandler.py]: I didn't understand. Can you elaborate this?

File: PatternFinding.py (hunk 5 above)
Comment (ujjaldas1997) [reply PatternFinding.py]: I didn't get it.

File: PatternFinding.py (hunk 6 above)
Comment (ujjaldas1997) [reply PatternFinding.py]: I think all indentations are correct. Did you encounter any error?

**Comment by ujjaldas1997** [comment]
I am really sorry. I didn't reviewed the code and opened a pull request. Now I have updated them.

File: Imagehandler.py (hunk 2 above)
Comment (Griffintaur) [reply Imagehandler.py]: is indentation correct?

File: Imagehandler.py (hunk 4 above)
Comment (Griffintaur) [reply Imagehandler.py]: unable

File: PatternFinding.py (hunk 5 above)
Comment (Griffintaur) [reply PatternFinding.py]: cv. destroyAllWinddows doesn't seem to be properly indented

**Comment by ujjaldas1997** [comment]
@Griffintaur It seems like you want me to include `cv.destroyAllWindows()` in the loop where `cv.imshow()`
relies. I have updated that. But only one  `cv.destroyAllWindows()` is enough to deallocate any associate memory, so calling `cv.destroyAllWindows()` outside the loop also works fine.

**Review by Griffintaur — APPROVED** [review]

**Comment by Griffintaur** [comment]
fixes #14

**Pull request merged**